sdk/azcore/response.go

```diff
@@ -120,12 +139,17 @@ func (r *Response) Drain() {
 }
 
 // removeBOM removes any byte-order mark prefix from the payload if present.
-func (r *Response) removeBOM() {
+func (r *Response) removeBOM() error {
```

Comment:
I'd add a comment like `// NOTE: you MUST call payload() before calling this method` as the below type assertion could fail in some cases.

Reply:
Done

Reply:
Actually, looking at the code, is the comment true? We call payload before trimming anyways.

Reply:
Derp you are correct, comment is redundant.

Reply:
Done, removed the comment.